src/babashka/process.cljc

```diff
@@ -178,17 +178,30 @@
       (str/includes? "windows")))
 
 (defn- -program-resolver [{:keys [program dir]}]
-  ;; this should make life easier and not cause any bugs that weren't there previously
-  ;; on exception we just return the program as is
-  (try
-    (if (fs/relative? program)
-      (if-let [f (fs/which (if dir
-                             (-> (fs/file dir program) fs/absolutize)
-                             program))]
-        (str f)
-        program)
-      program)
-    (catch Throwable _ program)))
+  ;; Used by default:
+  ;; - for all program resolution when running on Windows
+  ;; - by `exec` on all OSes
+  ;; Default ProcessBuilder behaviour on macOS/Linux does this work naturally
+  ;; and is therefore unneeded.
+  ;;
+  ;; ProcessBuilder program resolution on Windows does not entirely match
+  ;; behaviour of CMD Shell nor PowerShell.
+  ;; Here we adapt resolution to match PowerShell (with the exclusion of resolving
+  ;; .ps1 scripts) which follows the same strategy as macOS/Linux.
+  (if-let [resolved (cond
+                      (fs/absolute? program)
+                      (fs/which program) ;; to resolve any extensions
+
+                      (fs/parent program)
+                      (if dir
+                        ;; we need to absolutize here to overcome a bug in Windows ProcessBuilder
```

Comment:
Since this is also used in exec for non-Windows, should we check for windows here?

Reply:
From [comment](https://github.com/babashka/process/pull/165#issuecomment-2253202562) above:

> The absolutize is only necessary for ProcessBuilder on Windows, but should not be triggered for supported use of `exec` as `:dir` is not supported for `exec`.

Reply:
d'oh, ok then :)